realm/realm-library/src/androidTest/java/io/realm/RealmListTests.java

```diff
@@ -1019,4 +1022,167 @@ public void testRemoveAllFromRealmInvalidListShouldThrow() {
             assertEquals("This Realm instance has already been closed, making it unusable.", e.getMessage());
         }
     }
+
+    @Test
+    public void add_set_objectFromOtherThread() {
+        final CountDownLatch finishedLatch = new CountDownLatch(1);
+        final Dog dog = testRealm.where(Dog.class).findFirst();
+        final String expectedMsg = "Cannot copy an object from another Realm instance.";
```

Comment:
Shouldn't this throw an IllegalState because of a wrong thread first? Then check for this later?